cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -321,17 +321,11 @@ public TimeValuePair getLastCache(PartialPath seriesPath) {
   @Override
   public MNode getSeriesSchemasAndReadLockDevice(InsertPlan plan) throws MetadataException {
     MeasurementMNode[] measurementMNodes = new MeasurementMNode[plan.getMeasurements().length];
-    try {
-      MNode deviceNode = getDeviceNode(plan.getDeviceId());
-
-      int nonExistSchemaIndex =
-          getMNodesLocally(plan.getDeviceId(), plan.getMeasurements(), measurementMNodes);
-      if (nonExistSchemaIndex == -1) {
-        plan.setMeasurementMNodes(measurementMNodes);
-        return deviceNode;
-      }
-    } catch (PathNotExistException exception) {
-      // ignore, so we could try local MTree
+    int nonExistSchemaIndex =
+        getMNodesLocally(plan.getDeviceId(), plan.getMeasurements(), measurementMNodes);
+    if (nonExistSchemaIndex == -1) {
+      plan.setMeasurementMNodes(measurementMNodes);
+      return new MNode(null, plan.getDeviceId().getDevice());
```

Comment:
new MNode, which could add to cache?   Does the memory overhead increase if new is used each time?

Reply:
In cluster module, the return value of this function is not cached and is only used to detect whether an exception will be thrown.
In addition, an MNode instance is not very memory intensive, and the variables are null by default, so the overhead in memory should not be very large